### PR TITLE
Add bbe for csv `parseList` functionality

### PR DESCRIPTION
**Pull request by SasinduDilshara**
Fixes https://github.com/ballerina-platform/ballerina-dev-website/issues/9457

### CODE REVIEW

File: examples/parse-csv-lists/parse_csv_lists_to_ballerina_types.bal [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+public function main() returns error? {
+    // Sample CSV string array representing employee data.
+    string[][] csvList = [
+        ["1", "John", "Engineering", "1000.0"],
+        ["2", "Doe", "HR", "2000.0"],
+        ["3", "Jane", "Finance", "3000.0"]
+    ];
+
+    // Parse the CSV list into an array of `Employee` records by specifying custom headers.
+    Employee[] employees = check csv:parseList(csvList, {customHeaders: ["empId", "empName", "department", "salary"]});
```
Comment (heshanpadmasiri) [inline examples/parse-csv-lists/parse_csv_lists_to_ballerina_types.bal:32]: Is there a reason why we `check` error in this PR and in some others (#5668,  #5669, etc. ) we take both values? (I  personally prefer `check` unless you are doing something with errors)

Comment (SasinduDilshara) [reply examples/parse-csv-lists/parse_csv_lists_to_ballerina_types.bal:32]: Yeah, using `check` will be the better option

File: examples/parse-csv-lists/parse_csv_lists_to_ballerina_types.metatags [hunk 2]
```diff
@@ -0,0 +1,2 @@
+description: This BBE demonstrates the parsing of CSV lists into Ballerina records and arrays, both with and without data projection.
+keywords: ballerina, ballerina by example, bbe, csv, csv lists, record, record array, parseList, csv data module, data.csv, data projection
```
Comment (gimantha) [inline examples/parse-csv-lists/parse_csv_lists_to_ballerina_types.metatags:2]: csv to records

Comment (gimantha) [inline examples/parse-csv-lists/parse_csv_lists_to_ballerina_types.metatags:2]:
```suggestion
keywords: ballerina, ballerina by example, bbe, csv, csv lists, record, record array, parseList, csv data module, data.csv, data projection, csv to record array
```

File: examples/parse-csv-lists/parse_csv_lists_to_ballerina_types.bal (hunk 1 above)
Comment (SasinduDilshara) [reply examples/parse-csv-lists/parse_csv_lists_to_ballerina_types.bal:32]: Updated

**Pull request merged**